enhancements/cluster-logging/cluster-logging-es-rollover-data-design.md

```diff
@@ -0,0 +1,253 @@
+of indices maintained by the Elasticsearch server.  Each index can additionally be sharded to spread the load across the Elasticsearch nodes.  The end
+result of this shard explosion is the Elasticsearch cluster performance is not optimized and is not capable of efficiently processing logs or storing as many as it could. 
+
+Each index adds load and overhead (e.g. mapping, metadata) to the Elasticsearch cluster that needs to be tracked beyond the actual data.   Elasticsearch has recommendations for maximum shard size and the number of shards per node per allocated gigabyte of heap.  Cluster logging typically exceeds these recommendations for any OpenShift clusters that has significant log traffic.
+
+
+### Goals
+
+
+The goals of this proposals are:
+* Utilize a data design that aligns data schema with Elasticsearch's recommendations.
+* Expose data management policy as API in the `cluster-logging-operator` and `elasticsearch-operator` in support of Cluster Logging's mission to gather container and infrastructure logs
+ 
+
+### Non-Goals
+
+This change will not:
+* Provide a general data management policy API that fully exposes Elasticsearch's rollover API
+
+## Proposal
+
+This proposal introduces two specific changes to to achieve its goals:
+
+* Co-located data in a few opinionated indices
+* Index management using [rollover index API](https://www.elastic.co/guide/en/elasticsearch/reference/6.3/indices-rollover-index.html)
+
+Logs of a given type (e.g. app container, infra container, node) are separated by index.  The Cluster Logging collector writes logs to a well-known alias established by the `cluster-logging-operator`.  The ClusterLogging CR instance specifies the management policy for each log type index.  This controls: 
+* maximum number of logs per index (e.g. 30G) 
+* the maximum age (e.g. 7 days).  
+
+This policy is passed to the Elasticsearch CR for the `elasticsearch-operator` to manage the rollover policy.
+
+
+### Implementation Details
+
+#### Assumptions
+
+* ClusterLogging will expose the minimal needed set of Elasticsearch rollover policy management API in order to achieve the previously described goals
+* ClusterLogging will manage rollover as Elasticsearch index management is either restricted by Elastic licensing or not available in the opensource version of OpenDistro
+* Security will be addressed by using the OpenDistro security plugin and document level security (DLS). Details TBD.
+
+#### Data Model
+Logs of a given type are co-located to the following indices:
+
+| Log Type | Read Alias | Write Alias | Initial Index |
+|---------|-----------|---------------|---------------|
+| Node (`logs.infra`)|infra, infra.journal|infra.journal-write|infra.journal-00001|
+| Container Infra (`logs.infra`)|infra, infra.container|infra.container-write|infra.container-000001|
+| Container Application (`logs.app`)|app.logs|app.container-write|app.container-000001|
```

Comment:
where do audit logs go?

Reply:
Is it reasonable that audit logs should be separated from infra?

Reply:
i think so.....we're treating them separately for forwarding, right?

Reply:
It certainly is reasonable. They tend to have different usage and even data retention policy than infra logs.

Reply:
added audit logs to data model